packages/frontend/src/ee/pages/AiAgents/ProjectAiAgentEditPage.tsx

```diff
@@ -176,6 +196,83 @@ const ProjectAiAgentEditPage: FC<Props> = ({ isCreateMode = false }) => {
         [groups],
     );
 
+    const userOptions = useMemo(
+        () =>
+            usersWithProjectRole
+                // Filter out users who cannot view/interact with ai agents
+                // Filter out users that have default access to the ai agent - admins and developers
+                ?.filter((u) => {
+                    const projectRole = convertOrganizationRoleToProjectRole(
+                        u.finalRole,
+                    );
+                    const canCreateAiAgents =
+                        projectRole === ProjectMemberRole.ADMIN ||
+                        projectRole === ProjectMemberRole.DEVELOPER;
+                    const canInteractWithAiAgents =
+                        projectRole &&
+                        ProjectRoleOrder[projectRole] >=
+                            ProjectRoleOrder[
+                                ProjectMemberRole.INTERACTIVE_VIEWER
+                            ] &&
+                        !canCreateAiAgents;
+                    return canInteractWithAiAgents;
+                })
```

Comment:
question to discuss. - should I filter out current role or highest inherited role?

Reply:
100% - on it!